ddtrace/internal/runtime/lightstep_metrics.py

```diff
@@ -0,0 +1,215 @@
+            system_cpu_nice_total = self.cpu().nice
+            system_cpu = system_cpu_total - self.stored_value["SYSTEM_CPU_TOTAL"]
+            system_cpu_user = system_cpu_user_total - self.stored_value["SYSTEM_CPU_USER_TOTAL"]
+            system_cpu_idle = system_cpu_idle_total - self.stored_value["SYSTEM_CPU_IDLE_TOTAL"]
+            system_cpu_nice = system_cpu_nice_total - self.stored_value["SYSTEM_CPU_NICE_TOTAL"]
+
+            net_recv_total = self.net().bytes_recv
+            net_sent_total = self.net().bytes_sent
+            net_recv = net_recv_total - self.stored_value["NET_RECV_TOTAL"]
+            net_sent = net_sent_total - self.stored_value["NET_SENT_TOTAL"]
+
+            system_memory = self.mem()
+
+            self.previous_value = self.stored_value
+
+            self.stored_value = dict(
+                CPU_TIME_SYS_TOTAL=cpu_time_sys_total,
+                CPU_TIME_USER_TOTAL=cpu_time_user_total,
+                SYSTEM_CPU_TOTAL=system_cpu_total,
+                SYSTEM_CPU_USER_TOTAL=system_cpu_user_total,
+                SYSTEM_CPU_IDLE_TOTAL=system_cpu_idle_total,
+                SYSTEM_CPU_NICE_TOTAL=system_cpu_nice_total,
+                NET_RECV_TOTAL=net_recv_total,
+                NET_SENT_TOTAL=net_sent_total,
+            )
+
+            metrics = [
+                # process metrics
+                (LS_PROCESS_CPU_TIME_SYS, cpu_time_sys, MetricKind.COUNTER),
+                (LS_PROCESS_CPU_TIME_USER, cpu_time_user, MetricKind.COUNTER),
+                (LS_PROCESS_MEM_RSS, self.proc.memory_info().rss, MetricKind.GAUGE),
+                # system CPU metrics
+                (LS_SYSTEM_CPU_TIME_SYS, system_cpu, MetricKind.COUNTER),
+                (LS_SYSTEM_CPU_TIME_USER, system_cpu_user, MetricKind.COUNTER),
+                (LS_SYSTEM_CPU_TIME_IDLE, system_cpu_idle, MetricKind.COUNTER),
+                (LS_SYSTEM_CPU_TIME_NICE, system_cpu_nice, MetricKind.COUNTER),
+                # system memory metrics
+                (LS_SYSTEM_MEM_AVAIL, system_memory.available, MetricKind.GAUGE),
+                (LS_SYSTEM_MEM_USED, system_memory.used, MetricKind.GAUGE),
+                # system network metrics
+                (LS_SYSTEM_NET_RECV, net_recv, MetricKind.COUNTER),
+                (LS_SYSTEM_NET_SENT, net_sent, MetricKind.COUNTER),
+            ]
+
+            return metrics
+
+    def rollback(self):
+        self.stored_value = self.previous_value
+
+
+class LightstepRuntimeMetrics(RuntimeCollectorsIterable):
+    ENABLED = GC_RUNTIME_METRICS | LS_RUNTIME_METRICS
+    COLLECTORS = [
+        GCRuntimeMetricCollector,
+        LightstepPSUtilRuntimeMetricCollector,
+    ]
+
+    def rollback(self):
+        for c in self._collectors:
+            if hasattr(c, "rollback"):
+                c.rollback()
+
+
+class LightstepMetricsWorker(_worker.PeriodicWorkerThread):
+    """ Worker thread to collect and write metrics to a Lightstep endpoint """
+
+    FLUSH_INTERVAL = 5
```

Comment:
Is there a requirement to use all caps in these class attributes?

Reply:
It is not, updated.